api/oc_core_res.c

```diff
@@ -695,6 +701,10 @@ oc_core_is_vertical_resource(const oc_resource_t *resource, size_t device)
     }
   }
 
+  if (device >= OC_ATOMIC_LOAD32(g_device_count)) {
+    return false;
```

Comment:
add call to `CoreGetResourceIsVerticalResource_P`

Reply:
added